torch/lib/THD/base/channels/DataChannelTCP.cpp

```diff
@@ -196,70 +196,99 @@ DataChannelTCP::~DataChannelTCP()
 
 
 void DataChannelTCP::listen(std::uint16_t port = 0) {
-  SYSCHECK(_socket = ::socket(PF_INET, SOCK_STREAM, 0))
+  struct addrinfo hints, *res = NULL;
+
+  memset(&hints, 0x00, sizeof(hints));
+  hints.ai_flags = AI_PASSIVE;
+  hints.ai_family = AF_UNSPEC; // either IPv4 or IPv6
+  hints.ai_socktype = SOCK_STREAM; // TCP
+
+  // `getaddrinfo` will sort addresses according to RFC 3484 and can be tweeked
```

Comment:
The problem arise when we want to change this file (`/etc/gai.conf`) on another system different than Linux because it usually does not exists or exists under a different name. Not sure how it is implemented in OSX or other UNIX system. So I am not 100% about abandoning IPv6 preference.

So if we want still have IPv6 preference over IPv4 I can change that in a minute. But as I pointed out then we give up possibly better sorting to greedy picking.